### ISSUE TITLE
Cannot get DS4 (CUH-ZCT2U) to pair

### ISSUE THREAD

**Issue by Kiplar** [TheOfficialFloW/MiniVitaTV#17]
I have followed the instructions on the main MiniVitaTV page, and all I can get is a headphone icon or "Wireless Controller". I don't tap on them as the instructions have said, and under Bluetooth Feature it says "Not connected. There are no registered Bluetooth devices." I try to hit the PS Button on the controller so it can sync, but that will just cancel the pairing. (SHARE + PS Button).

If I tap "Wireless Controller" when it is trying to pair, it will fail. But then if I back out of Bluetooth Devices and then go back in, I will have an option that says "DUALSHOCK 4 Wireless Controller" but it still will not pair. Light will just keep on blinking.
If I tap "DUALSHOCK 4 Wireless Controller" it will tell me "To connect, press any button on the Bluetooth device." Which does not result in it connecting. Pressing PS Button still cancels the pairing operation on Controller. On the Vita, the only other options I get in this menu are Delete, and a grayed out Connect button.

What am I doing wrong here? Is it specific to the CUH-ZCT2U version of the Dualshock 4?

**Comment by Kiplar** [comment]
Still having immense troubles with this. I even got the Dualshock 3 to connect, but I simply cannot get them to connect at the same time.

**Comment by amina-dev** [comment]
When using this plugin, make sure that your Bluetooth is ON before powering up your Vita so that you will not experience that error when connecting your DS4 to your device.

**Comment by maskmark** [comment]
ehh..actually today I even paired a 3rd party ds4 controller and it worked perfectly with the plugin.

**Comment by RitschiRitsch** [comment]
I've had the same issue. Try the following:

    Do not use ds4vita (i.e. delete ds4vita.skprx in your config.txt).
    Go to settings, devices, bluetooth devices, delete paired devices. Leave bluetooth turned on.
    Install MiniVitaTV as stated in the readme.
    Reboot
    (N: only touchscreen is working) on your vita go to settings, devices, bluetooth devices. Then press the share + PS button on your DS4 controller to enter into pairing mode.
    Press "connect" on the vita when prompted.
    If you get an error and your connection to your DS4 will just abort as soon as you turn on the DS4 by pressing the PS button (i.e. the lightbar just turns on for a sec and then the controller turns off immediately), then just reboot your vita by shutting it down via the power button and then power it back on. After that reboot, just press the PS button of your DS4 and it will then connect automatically. Make sure to leave bluetooth enabled.

Hence, this is not really an issue. It is more an inconvenient way to get it working :-)

**Comment by RuiOrey** [comment]
To me it never appears connect prompt and appears always "Wireless Controller" with profile error if I click on it. Is this compatible with all versions of ds4?

**Comment by RitschiRitsch** [comment]
I was able to connect with a V1 and V2 DS4. Make sure to delete paired devices. Then try connecting. If your DS4 just turns off during pairing and after that if you turn on your DS4, it immediately turns off, then just reboot your vita (with power button or in Henkaku settings). Keep bluetooth always enabled. After reboot just press the PS button on your DS4 and it should connect automatically. No need to go into bluetooth settings anymore.

**Comment by RuiOrey** [comment]
During pairing it just appears as "Wireless Controller" and does not do
anything. If I select to connect, it says it has a wrong profile.
The only way I got it to appear as DS4 but without responding was when I
switched ds4vita to MiniVitaTV and restarted without unpairing the
controller.
Other than that it never connects. I'm using 3.6

RitschiRitsch <notifications@github.com> escreveu no dia sábado, 2/05/2020
à(s) 16:09:

**Comment by RitschiRitsch** [comment]
Do not use ds4vita and MiniVitaTV plugins at the same time. You would only need minivitatv.skprx and ds3.skprx (yes you need both of them, regardless whether you use a DS3 or DS4). Make sure to disable ds4vita in your config.txt.

**Comment by RuiOrey** [comment]
> Do not use ds4vita and MiniVitaTV plugins at the same time. You would only need minivitatv.skprx and ds3.skprx (yes you need both of them, regardless whether you use a DS3 or DS4). Make sure to disable ds4vita in your config.txt.

Yes, they are disabled when I enable miniVitaTV. What I meant was that the ONLY WAY I got the controller to be recognized when minivita was enabled, was to previously pair it in ds4vita and then disable ds4vita, enable MiniVitaTV and then restarting. However in that way the controller appears connected on the bluetooth setting page as a DS$, the controller light is fixed, but the controller does not respond on the vita.


> I was able to connect with a V1 and V2 DS4. Make sure to delete paired devices. Then try connecting. If your DS4 just turns off during pairing and after that if you turn on your DS4, it immediately turns off, then just reboot your vita (with power button or in Henkaku settings). Keep bluetooth always enabled. After reboot just press the PS button on your DS4 and it should connect automatically. No need to go into bluetooth settings anymore.

Also not working. When you say "If your DS4 just turns off during pairing" you mean while seeing him appearing as a "Wireless controller" on the bluetooth settings?

Thanks a lot for the help

**Comment by RuiOrey** [comment]
By the way this is my `config.txt`

```
# This file is used as an alternative if ux0:tai/config.txt is not found.
# For users plugins, you must refresh taiHEN from HENkaku Settings for
# changes to take place.
# For kernel plugins, you must reboot for changes to take place.
*KERNEL
ur0:tai/storagemgr.skprx
#ur0:tai/ds4vita.skprx
#ur0:tai/ds3vita.skprx
ur0:tai/nonpdrm.skprx
ur0:tai/nopsmdrm.skprx
ur0:tai/psonescrot.skprx
ur0:tai/reF00D.skprx
ur0:tai/udcd_uvc_oled_off.skprx
ur0:tai/repatch.skprx
ur0:tai/minivitatv.skprx
ur0:tai/ds3.skprx
ux0:app/PSPEMUCFW/sce_module/adrenaline_kernel.skprx
# henkaku.skprx is hard-coded to load and is not listed here
*main
ur0:tai/download_enabler.suprx
ur0:tai/pngshot.suprx
ur0:tai/quietshot.suprx
ur0:tai/nolockscreen.suprx
ur0:tai/quick_power.suprx
ur0:tai/shellbat_mod.suprx
ur0:tai/ftpeverywhere.suprx
# main is a special titleid for SceShell
ur0:tai/henkaku.suprx
*NPXS10015
# this is for modifying the version string
ur0:tai/henkaku.suprx
*NPXS10016
# this is for modifying the version string in settings widget
ur0:tai/henkaku.suprx
*ALL
ur0:tai/PSMPatch.suprx
ur0:tai/FuckPSSE.suprx
ur0:tai/InfiniteNet.suprx
*NPXS10095
ur0:tai/quietshot.suprx
```

**Comment by RitschiRitsch** [comment]
Try to place the minivita and ds3 plugins in ux0:tai (not ur0:tai) and edit the config accordingly. Maybe they have to be in ux0 because of dependencies of each other. Unfortunately TheFl0w did not release the source code of ds3.skprx.

**Comment by RuiOrey** [comment]
> Try to place the minivita and ds3 plugins in ux0:tai (not ur0:tai) and edit the config accordingly. Maybe they have to be in ux0 because of dependencies of each other. Unfortunately TheFl0w did not release the source code of ds3.skprx.

Ok after: 
- moving the files to ux0 and uma0
- factory resetting the controller
- selecting the option to connect to "Wireless Controller" on BT options
it is working!

 All the triggers and rumble. Great. Thanks a lot!

Maybe the information that files can only be on ux0 should be on the README.MD

**Comment by RuiOrey** [comment]
Also I'm not sure if the analogs are working. Are they supposed to work?

**Comment by RitschiRitsch** [comment]
You‘re welcome. However, it is exactly stated in the readme to put the files in ux0 and nowhere else. So the readme is fine. You just have to stick to the instructions strictly. Happy dualshocking :)

**Comment by RuiOrey** [comment]
Nice to know about the analogs.

Allow me to disagree a bit on you in terms of how clear the ux0 reference is for someone who is hacking the vita. I don't think is only a matter of lack of attention.

What I mean is something like we have this:

**Only official DS3/DS4 controllers are currently supported!**

like 
**For SD2Vita and similar alternative storage users: This plugin to work must me placed specifically on ux0:/tai and cannot be put on other drive like ur0: or uma0:**

On the official documentation, it says to be put on `ux0:\tai` but is not emphasized. Usually in instructions of SD2Vita (for example) we are told to move every plugin to the `ur0:\tai`.
In principle it works.
Instructions for `ds4vita` for example, they say to put also on `uxo` but it also works on `ur0`

Because I think usually all plugins are to be put on the tai folder of the drive that we are using as main and is hard

**Comment by RuiOrey** [comment]
Sorry about the editions. pressed enter before finishing writing... my bad!

**Comment by RuiOrey** [comment]
Ok just to let know that I removed both ux0:\tai and uma0:\tai and minivitatv is now working corretly from ur0:\tai ! So it seems is not mandatory to be on ux. Thanks anyaway @RitschiRitsch !
